### ISSUE TITLE
First element in an anonymous code block (initializer) shows as a syntax error.

### ISSUE THREAD

**Issue by ericfillipe1** [HaxeFoundation/intellij-haxe#93]
![haxe](https://cloud.githubusercontent.com/assets/4976223/5554838/4398e45c-8c61-11e4-86ee-79500a83ebeb.png)

TIR: STB-7313

**Comment by EBatTiVo** [comment]
It looks like the plugin only allows structure initialization in an initializer block.  (e.g. it accepts ```{ a: "Test"; b: 1; }, but not random code, though the latter is legal Haxe syntax.

**Comment by Simn** [comment]
Parsing expression-level `{` correctly can be a bit tricky. The problem is that you can't distinguish expression blocks from object declarations until you find the `:`. Both `{ foo` and `{ "foo"` could be an object declaration such as `{ foo: 1 }` and `{ "foo": 1 }`, but it could also be an expression block such as `{ foo.call(); }` and `{ "foo".call(); }`.

Our parser has two extra states block1 and block2 such that:

```
"{" ::= block1
block1 ::= ident block2 | string block2 | block
block2 ::= ":" parse_obj_decl | parse_expr_next
```

Apologies for the crude BNF notation, I hope it gets the idea across.

**Comment by EricBishton** [comment]
Does not occur in v0.11.2.  Probably fixed during the 0.11.0 development cycle.
